pkgs/test/named-hierarchy/reference/src/main.rs

```diff
@@ -0,0 +1,147 @@
+use anyhow::Context;
+use std::ffi::OsStr;
+use std::collections::HashSet;
+use anyhow::Error;
+use std::path::PathBuf;
+use rnix::SyntaxKind::NODE_PATH;
+use rowan::ast::AstNode;
+use rnix::Root;
+use std::fs::read_to_string;
+use anyhow::Result;
+use std::env;
+
+mod line_index;
+use line_index::LineIndex;
+
+use std::path::Component;
+
+fn resolve_reference(
+    source: &PathBuf,
+    line: usize,
+    reference: PathBuf,
+    base_dir: &PathBuf,
+) -> Result<PathBuf> {
+    let mut rel_to_root = source.parent().unwrap().to_path_buf();
```

Comment:
Won't work as is, PathBuf::parent returns Option and this function returns Result.

Reply:
This condition can never occur because this function is only called with files, which always have a parent. Because of this I think `unwrap` is okay here, but I'll add a comment justifying why